contrib/filter-repo-demos/lint-history

```diff
@@ -122,22 +130,26 @@ def lint_with_real_filenames(commit, metadata):
       cat_file_process.stdin.flush()
       objhash, objtype, objsize = cat_file_process.stdout.readline().split()
       contents_plus_newline = cat_file_process.stdout.read(int(objsize)+1)
-
-      # Write it out to a file with the same basename
-      filename = os.path.join(tmpdir, os.path.basename(change.filename))
-      with open(filename, "wb") as f:
-        f.write(contents_plus_newline[:-1])
-
-      # Lint the file
-      subprocess.check_call(lint_args.command + [filename.decode('utf-8')])
-
-      # Get the new contents
-      with open(filename, "rb") as f:
-        blob = fr.Blob(f.read())
+      
+      if lint_args.callback:
+        blob = fr.Blob(contents_plus_newline)
+        callback(change,blob)
```

Comment:
I don't understand; what good can this do since blob will still refer to the original value?  I would have thought I'd see a `blob = callback...`; am I missing something?